code/modules/mob/living/simple_animal/hostile/hostile.dm

```diff
@@ -112,7 +112,11 @@
 			if(can_see(targets_from, HM, vision_range))
 				. += HM
 	else
-		. = oview(vision_range, targets_from)
+		. = list() // The following code is only very slightly slower than just returning oview(vision_range, targets_from), but it saves us much more work down the line, particularly when bees are involved
+		for (var/obj/A in oview(vision_range, targets_from))
+			. += A
```

Comment:
You could do 
```
for(var/atom/movable/A in oview(...))
    if(A.type == /atom/movable/lighting_object)
        continue
    . += A
```

Reply:
Though i'm not sure if that's faster

Reply:
Subtypes are not "accelerated" for `for (var/type/A in oview(foo,bar))` by the compiler. This is a special case and not just me doing this for shits and giggles.

Reply:
`atom/movable` might be accelerated since it's a special byond construct.

Reply:
I started wondering if it might actually work, and put this change in my profiling branch. Unfortunately, it's 4.3% slower, so I don't think the compiler is working its magic on it.

Reply:
Update: Cyberboss's way is between 0.6% faster and 0.4% slower, probably just down to whatever mobs got picked for which scheme